packages/@aws-cdk/aws-ec2/lib/vpc-ref.ts

```diff
@@ -227,6 +239,20 @@ export abstract class VpcNetworkBase extends Construct implements IVpcNetwork {
     }[placement.subnetsToUse];
   }
 
+  /**
+   * Returns IDs of selected subnets
+   */
+  public subnetIds(placement: VpcPlacementStrategy = {}): string[] {
+    placement = reifyDefaultsPlacement(placement);
+
+    const nets = this.subnets(placement);
+    if (nets.length === 0) {
+      throw new Error(`There are no ${describePlacement(placement)} in this VPC. Use a different VPC placement strategy.`);
```

Comment:
Could we maybe hint at what strategies would have matched something? Or are placement strategies too complex?

Reply:
They are not necessarily, but what are you proposing?

`"The VPC network contains 1 Public subnet group with name "Public" and 2 Private subnets groups with names "MyDatabases","MyLambdas"`

Reply:
I mean I wouldn't put the `Use a different VPC placement strategy` if you can't hint at which are "valid".

Reply:
I used that terminology on purpose because:

* It's okay to state an error, but it's better to state a path to a solution.
* I want to use the words "VPC placement" here on purpose, because previously asked questions have shown that people don't think of this term.

Although maybe the better solution is to change the parameter name to `subnetSelection`, or something, under the "s" where people that already know things about AWS and CFN are going to look for it.